pymc3/backends/arviz.py

```diff
@@ -0,0 +1,614 @@
+        if self.ndraws > 0:
+            trace_posterior = self.trace[self.ntune :]
+        return trace_posterior, trace_warmup
+
+    @requires("trace")
+    def posterior_to_xarray(self):
+        """Convert the posterior to an xarray dataset."""
+        var_names = get_default_varnames(self.trace.varnames, include_transformed=False)
+        data = {}
+        data_warmup = {}
+        for var_name in var_names:
+            if self.warmup_trace:
+                data_warmup[var_name] = np.array(
+                    self.warmup_trace.get_values(var_name, combine=False, squeeze=False)
+                )
+            if self.posterior_trace:
+                data[var_name] = np.array(
+                    self.posterior_trace.get_values(var_name, combine=False, squeeze=False)
+                )
+        return (
+            dict_to_dataset(
+                data,
+                library=pymc3,
+                coords=self.coords,
+                dims=self.dims,
+                attrs=self.attrs,
+                index_origin=self.index_origin,
+            ),
+            dict_to_dataset(
+                data_warmup,
+                library=pymc3,
+                coords=self.coords,
+                dims=self.dims,
+                attrs=self.attrs,
+                index_origin=self.index_origin,
+            ),
+        )
+
+    @requires("trace")
+    def sample_stats_to_xarray(self):
+        """Extract sample_stats from PyMC3 trace."""
+        data = {}
+        rename_key = {
+            "model_logp": "lp",
+            "mean_tree_accept": "acceptance_rate",
+            "depth": "tree_depth",
+            "tree_size": "n_steps",
+        }
+        data = {}
+        data_warmup = {}
+        for stat in self.trace.stat_names:
+            name = rename_key.get(stat, stat)
+            if name == "tune":
+                continue
+            if self.warmup_trace:
+                data_warmup[name] = np.array(
+                    self.warmup_trace.get_sampler_stats(stat, combine=False)
+                )
+            if self.posterior_trace:
+                data[name] = np.array(self.posterior_trace.get_sampler_stats(stat, combine=False))
+
+        return (
+            dict_to_dataset(
+                data,
+                library=pymc3,
+                dims=None,
+                coords=self.coords,
+                attrs=self.attrs,
+                index_origin=self.index_origin,
+            ),
+            dict_to_dataset(
+                data_warmup,
+                library=pymc3,
+                dims=None,
+                coords=self.coords,
+                attrs=self.attrs,
+                index_origin=self.index_origin,
+            ),
+        )
+
+    @requires("trace")
+    @requires("model")
+    def log_likelihood_to_xarray(self):
+        """Extract log likelihood and log_p data from PyMC3 trace."""
+        # TODO: add pointwise log likelihood extraction to the converter
+        return None
```

Comment:
I have disabled the log likelihood conversion for now so we can already start working on tests that depend on this. The ones I am adding from ArviZ do test specifically for log likelihood conversion but I don't think the tests on pymc3 that use `sample` will. I hope this will already work.

Reply:
Sounds good!